numpyro/infer/elbo.py

```diff
@@ -932,10 +932,26 @@ def single_particle_elbo(rng_key):
                     # uncontracted logp cost term
                     assert len(group_names) == 1
                     name = next(iter(group_names))
-                    cost = model_trace[name]["log_prob"]
-                    scale = model_trace[name]["scale"]
-                    deps = model_deps[name]
-                    dice_factors = [guide_trace[key]["log_measure"] for key in deps]
+                    try:
+                        cost = -kl_divergence(
+                            guide_trace[name]["fn"], model_trace[name]["fn"]
+                        )
```

Comment:
Maybe move `try/except` logic to `get_importance_trace_enum` to avoid wasteful computation of `log_prob`s?

Reply:
are you sure this is correct? what are the conditions under which you can compute the KL? note the conditions i wrote down for the mean field elbo:

```
The precise condition is that for every
    latent variable `z` in the guide, its parents in the model must not include
    any latent variables that are descendants of `z` in the guide. Here
    'parents in the model' and 'descendants in the guide' is with respect
    to the corresponding (statistical) dependency structure. For example, this
    condition is always satisfied if the model and guide have identical
    dependency structures.
```